build-executor/src/main/java/org/jboss/pnc/executor/DefaultBuildExecutor.java

```diff
@@ -374,9 +374,17 @@ private Void completeExecution(DefaultBuildExecutionSession buildExecutionSessio
             buildExecutionSession.setStatus(BuildExecutionStatus.DONE);
         }
 
-        log.debug("Removing buildExecutionTask [" + buildExecutionSession.getId() + "] form list of running tasks.");
+        log.debug("Removing buildExecutionTask [" + buildExecutionSession.getId() + "] from list of running tasks.");
         runningExecutions.remove(buildExecutionSession.getId());
 
+        String accessToken = buildExecutionSession.getAccessToken();
+        log.debug("Closing Maven repository manager [" + buildExecutionSession.getId() + "].");
+        try {
+            repositoryManagerFactory.getRepositoryManager(ArtifactRepo.Type.MAVEN).close(accessToken);
+        } catch (ExecutorException executionException) {
+            e = executionException;
```

Comment:
This exception "e" is not handled down-here. See ` if (e != null) {` few lines above.